src/toil/jobStores/aws/jobStore.py

```diff
@@ -74,38 +82,70 @@ def copyKeyMultipart(srcKey, dstBucketName, dstKeyName, headers=None):
     :rtype: boto.s3.multipart.CompletedMultiPartUpload
     :return: An object representing the completed upload.
     """
+    def copyPart(partIndex):
+        if exceptions:
+            return None
+        else:
+            try:
+                for attempt in retry_s3():
+                    with attempt:
+                        # Note that part_num is indexed from 1 while start and end are indexed from
+                        # 0. Notice also that the partIndex variable is indexed from 0.
+                        part = upload.copy_part_from_key(src_bucket_name=srcKey.bucket.name,
+                                                         src_key_name=srcKey.name,
+                                                         src_version_id=srcKey.version_id,
+                                                         part_num=partIndex + 1,
+                                                         start=partIndex * partSize,
+                                                         end=min(partIndex * partSize + partSize,
+                                                                 totalSize) - 1,
+                                                         headers=headers)
+                log.debug('Successfully copied part %d of %d' % (partIndex, totalParts))
+
+            except Exception as e:
+                if partIndex <= 4:
+                    exceptions[partIndex] = traceback.format_exc()
+                else:
+                    exceptions[partIndex] = e
+                log.debug('Part number %s of copy failed.', partIndex)
+                return None
+            else:
+                return part
+
     partSize = defaultPartSize
+    totalSize = srcKey.size
+    totalParts = (totalSize + partSize - 1) / partSize
+    exceptions = collections.OrderedDict()
     # We need a location-agnostic connection to S3 so we can't use the one that we
     # normally use for interacting with the job store bucket.
     with closing(boto.connect_s3()) as s3:
-        headers = headers or {}
-        totalSize = srcKey.size
         for attempt in retry_s3():
             with attempt:
                 dstBucket = s3.get_bucket(dstBucketName)
                 upload = dstBucket.initiate_multipart_upload(dstKeyName, headers=headers)
+        log.info("Initiated multipart copy from 's3://%s/%s' to 's3://%s/%s'." %
+                 (srcKey.bucket.name, srcKey.name, dstBucketName, dstKeyName))
```

Comment:
https://github.com/BD2KGenomics/toil/wiki/Python-Dos-and-Don'ts#dont-use-format-or--to-format-log-messages